sunpy/map/mapbase.py

```diff
@@ -2747,19 +2753,42 @@ def contour(self, level, **kwargs):
          (719.58811599, -356.68119768), (721.29217122, -354.76448374),
          (719.59798458, -352.60839064)]>
 
+        >>> contours = aia.contour(0.99*u.DN, area=True)  # doctest: +REMOTE_DATA
+        >>> print(contours[0])  # doctest: +REMOTE_DATA
+        <SkyCoord (Helioprojective: obstime=2011-06-07T06:33:02.880, rsun=696000.0 km, observer=<HeliographicStonyhurst Coordinate (obstime=2011-06-07T06:33:02.880, rsun=696000.0 km): (lon, lat, radius) in (deg, deg, m)
+        (-0.00406429, 0.04787238, 1.51846026e+11)>): (Tx, Ty) in arcsec
+        [(100.16572879, -1143.25308903), ( 99.68139649, -1143.76979777),
+         (100.159315  , -1145.90002049), (100.68236775, -1143.77222046),
+         (100.16572879, -1143.25308903)]>
+    
         See Also
         --------
         skimage.measure.find_contours
         """
         from skimage import measure
-
+        
         level = self._process_levels_arg(level)
         if level.size != 1:
             raise ValueError("level must be a single scalar value")
         else:
             # _process_levels_arg converts level to a 1D array, but
             # find_contours expects a scalar below
             level = level[0]
+        if area:
+            if not 0 <= level <= 1:
+                # print(level)
+                raise ValueError('level must be between 0 and 1 when area is set to True.')
+            # Sort pixels by value, then find the value that corresponds to the
```

Comment:
Author attribution should be given here since I think this is basically the gist code from the original issue?